### PR TITLE
EFO version aware OnToma update

### PR DESCRIPTION
**Pull request by DSuveges**
- [x]  When creating the cache, ontoma saves the efo versions.
- [x] If next time the cache is read, it is made sure the version matches.
- [x] Slight improvement on the logic here and there.
- [x] Mapped terms only logged in debug. (it was polluting the logs.) 

**!Important!** No change in the mapping logic. See ticket for more information: [#2847](https://github.com/opentargets/issues/issues/2847)

### CODE REVIEW

File: ontoma/EFO_Handler.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+"""Class to handle EFO ontology related functions."""
```
Comment (tskir) [inline ontoma/EFO_Handler.py]: PEP8 says packages should have all-lowercase names (although underscores are permitted): https://peps.python.org/pep-0008/#package-and-module-names

File: ontoma/EFO_Handler.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+from retry import retry
+from typing import Dict
+
+from ontoma import ontology
+
+logger = logging.getLogger(__name__)
+
+
+class efo_handler:
+    """A collection of functions to manage EFO data retrieve, version control and caching."""
```
Comment (tskir) [inline ontoma/EFO_Handler.py]:
```suggestion
    """A collection of functions to manage EFO data retrieving, version control and caching."""
```

File: ontoma/EFO_Handler.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import requests
+from retry import retry
+from typing import Dict
+
+from ontoma import ontology
+
+logger = logging.getLogger(__name__)
+
+
+class efo_handler:
```
Comment (tskir) [inline ontoma/EFO_Handler.py]: While in rare cases PEP8 does allow this in rare cases, I think in our case it's best to use the mode widespread camelcase convension for this class name (`EFOHandler`)

File: ontoma/cli.py [hunk 4]
```diff
@@ -10,8 +11,15 @@
...
-    return open('../VERSION').read().strip()
+def get_version() -> str:
+    """
+    It reads the VERSION file in the same directory as the script and returns its contents
+
+    Returns:
+      The version number of the script.
+    """
+    script_path = os.path.dirname(__file__)
+    return open(f'{script_path}/../VERSION').read().strip()
```
Comment (tskir) [inline ontoma/cli.py]: Thank you for fixing this! I noticed that when calling `ontoma --version`, it doesn't add a newline. While we're at it, could you please fix this as well in the `cli.py`?

image

File: ontoma/EFO_Handler.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            for synonym in term.synonyms:
+                if synonym.description and synonym.scope == "EXACT":
+                    synonyms_dataset.append(
+                        [
+                            synonym.description.lower(),
+                            normalised_term_id,
+                        ]
+                    )
+
+        # Strore datasets:
```
Comment (tskir) [inline ontoma/EFO_Handler.py]:
```suggestion
        # Store datasets:
```

File: ontoma/cli.py [hunk 6]
```diff
@@ -10,8 +11,15 @@
...
 
-def get_version():
-    return open('../VERSION').read().strip()
+def get_version() -> str:
+    """
+    It reads the VERSION file in the same directory as the script and returns its contents
+
+    Returns:
+      The version number of the script.
+    """
```
Comment (tskir) [inline ontoma/cli.py]:
```suggestion
def get_version() -> str:
    """
    Reads the VERSION file in the same directory as the script and returns its contents.

    Returns:
        The version number of the script.
    """
```

**Review by tskir — CHANGES_REQUESTED** [review]
Thank you very much for this PR Daniel, great stuff.

I notice that there are many (I presume automated) style changes, such as single quotes to double quotes, or list formatting etc. Are you using the VS Code linter for this? (I don't have anything at all against this, just want to make sure we're using the same linter so that they don't start changing things back and forth.)

File: ontoma/EFO_Handler.py (hunk 1 above)
Comment (DSuveges) [reply ontoma/EFO_Handler.py]: I didn't put much though into the names. I'm happy to change.

**Comment by DSuveges** [comment]
> Are you using the VS Code linter for this? (I don't have anything at all against this, just want to make sure we're using the same linter so that they don't start changing things back and forth.)

Yes @tskir , I'm using VS Code. I am thinking about establishing a shared developer environment like we have with the genetics_etl repo. I'm adding the config files, however I'm not sure if this is enough to unify the formatting.

**Review by tskir — APPROVED** [review]

**Pull request merged**